Feature #1797 Database optimization v2

PaulBender opened the pull request:
All Database optimization from #1797

PostgreSQL and SQLite don't support unsigned ints.

Destroy666x commented on install/resources/upgrade33.php:
```diff
@@ -0,0 +1,107 @@
+			$db->modify_column("usergroups", "type", "tinyint NOT NULL default '2'");
+			break;
+		default:
+			$db->modify_column("usergroups", "type", "tinyint(2) unsigned NOT NULL default '2'");
```
PostgreSQL doesn't support tinyint.

Destroy666x commented on install/resources/upgrade33.php:
```diff
@@ -0,0 +1,107 @@
+	{
+		// PostgreSQL and SQLite do not support unsigned ints
+		case "sqlite":
+			$db->modify_column("usergroups", "type", "tinyint NOT NULL default '2'");
```
SQLite supports the number of digits in parentheses.

PaulBender commented:
Made all changes.
-Changed type to tinyint(1) unsigned
-Dropped all unsigned references in SQLite install file
-Set a default of 0 for uid, touid and date in the buddyrequests table

PaulBender commented:
Changed loginattempts to an unsigned smallint (http://community.mybb.com/thread-166156.html)

PaulBender commented:
Fixed merged conflicts
